File: two_city_scheduling/TwoCityScheduling.java
class TwoCityScheduling {
    public int twoCitySchedCost(int[][] costs) {
        if (costs == null || costs.length == 0) {
            return 0;
        }

        int totalCost = 0;
        for (int i = 0; i < costs.length; i++) {
            costs[i][1] -= costs[i][0];
            totalCost += costs[i][0];
        }
        Arrays.sort(costs, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                if (o1[1] == o2[1]) {
                    return 0;
                }
                return o1[1] < o2[1] ? -1 : 1;
            }
        });

        for (int i = 0; i < costs.length / 2; i++) {
            totalCost += costs[i][1];
        }

        return totalCost;
    }
}

File: two_city_scheduling/Two_City_Scheduling.cpp
class Two_City_Scheduling {    
public:
    int twoCitySchedCost(vector<vector<int>>& costs) {
        if (costs.empty()) {
            return 0;
        }

        int min_cost = 0;
        for (int i = 0; i < costs.size(); i++) {
            min_cost += costs[i][0];
            costs[i][1] -= costs[i][0];
        }

        sort(costs.begin(), costs.end(), [](vector<int>& cost1, vector<int>& cost2) {
            return cost1[1] < cost2[1];
        });
        for (int i = 0; i < costs.size() / 2; i++) {
            min_cost += costs[i][1];
        }
        return min_cost;
    }
};